code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -398,13 +400,13 @@
 	glass_desc = "All natural, antioxidant-rich flavour sensation."
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
-/datum/reagent/consumable/icetea/on_mob_life(mob/living/carbon/M)
-	M.dizziness = max(0,M.dizziness-2)
-	M.drowsyness = max(0,M.drowsyness-1)
-	M.AdjustSleeping(-40)
-	if(M.getToxLoss() && prob(20))
+/datum/reagent/consumable/icetea/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	M.dizziness = max(M.dizziness - (2 * REM * delta_time), 0)
+	M.drowsyness = max(M.drowsyness - (1 * REM * delta_time), 0)
+	M.AdjustSleeping(-40 * REM * delta_time)
+	if(M.getToxLoss() && DT_PROB(10, delta_time))
 		M.adjustToxLoss(-1, 0)
-	M.adjust_bodytemperature(-5 * TEMPERATURE_DAMAGE_COEFFICIENT, M.get_body_temp_normal())
+	M.adjust_bodytemperature(-5 * REM * TEMPERATURE_DAMAGE_COEFFICIENT * delta_time, M.get_body_temp_normal())
 	..()
 	. = 1
```

Comment:
You mind setting this to TRUE? I assume that's what it's meant to be

Reply:
No problem